helix-core/src/main/java/org/apache/helix/controller/stages/CustomizedViewAggregationStage.java

```diff
@@ -107,8 +114,11 @@ public void execute(final ClusterEvent event) throws Exception {
           }
           // add/update customized-views from zk and cache
           if (updatedCustomizedViews.size() > 0) {
-            dataAccessor.setChildren(keys, updatedCustomizedViews);
+            boolean[] success = dataAccessor.setChildren(keys, updatedCustomizedViews);
             cache.updateCustomizedViews(stateType, updatedCustomizedViews);
+            asyncReportLatency(cache.getAsyncTasksThreadPool(), getOrCreateMonitor(event),
+                new ArrayList<>(updatedCustomizedViews), curCustomizedViewsCopy,
```

Comment:
Have you run the whole integration tests? I'm a bit concerned about the extra memory cost. Do you need to whole curCustomizedViewCopy? The ones that are corresponding to the updatedViews should be fine, right? as you only need those timestamps to compare.

Reply:
Discussed offline, current curCustomizedViewCopy is a whole copy of the entire set of customized views. We do not need that, so I will only save the curCustomizedView of what's changed, so we won't need to be concerned about memory usage. So curCustomizedViewCopy will be initialized as an empty map, and when we compare the customized views, if it's changed, we put the old customized view into this map, for later use to find out which partition state changed.